Sources/TensorFlow/Layers/Embedding.swift

```diff
@@ -0,0 +1,63 @@
+// Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// An embedding layer.
+///
+/// `Embedding` is effectively a lookup table that maps indices from a fixed vocabulary to fix sized
+/// (dense) vector representations.
+/// E.g. `[[0], [3]] -> [[0.25, 0.1], [0.6, -0.2]]`
+public struct Embedding<Scalar: TensorFlowFloatingPoint>: Layer {
+    /// A helper structure containing the indices to be passed into the `Embedding` layer.
+    ///
+    /// - Note: Often times, `Embedding` is followed by a `Flatten` and a `Dense` layer. When this 
+    ///   is the case, ensure that all input sequences of indices have the same dimension.
+    /// - Note: This structure is needed to conform `Embedding` to the Layer protocol.
```

Comment:
This particular note is an implementation note (not part of API documentation comments), so it should use two slashes.
```suggestion
    // NOTE: This structure is needed to conform `Embedding` to the Layer protocol.
```